### ISSUE TITLE
customize location for each project on one-repo mode

### ISSUE THREAD

**Issue by kennir** [IvanMalison/org-projectile#5]
How can I customize the location for each project? I have many projects are same name,so the location also be confused together

**Comment by IvanMalison** [comment]
Which Todo storage system are you using? Single file, per repo, hybrid?

**Comment by IvanMalison** [comment]
Oh oops, just saw that you are using single file mode. Unfortunately, as of
right now, there is not really a way to do this. It is sort of assumed that
project names will be unique. There are potentially some ways this could be
fixed. Perhaps uniquefy could be applied to project names and the
containing directory would be used to identify the project. Alternatively,
I could add an option that would identify projects by their full path a.

**Comment by kennir** [comment]
Yeah, I'm using single file mode, I can't put a irrelevant file to my project folder,  maybe use full path to identify projects is ok . 

I'd really like this plugin, It's very useful, thank you for your job

**Comment by IvanMalison** [comment]
Thats good yo hear; I'm glad that other people are using it. I can make it
so that you can supply a custom function to generate the heading from the
filepath. In your case you would just supply an identity function to get
full filepath headings.

**Comment by IvanMalison** [comment]
Hmmm. This is actually a bit harder to fix than I originally thought it was. It's going to require that I change the underlying way that org-projectile deals with projects so that everything is done with filepaths (probably for the best anyway).

**Comment by kennir** [comment]
Ok,Maybe you can add this feature on next major updating :)

**Comment by IvanMalison** [comment]
Yeah. I'd definitely like to do this, but it actually seems sort of hard, I want to be able to let people continue to simply use project names to identify their projects. This is important for me because i use org-projectile across multiple computers where the projects are often found at different filepaths. I'm not really sure what the right way to approach the issue is.

**Comment by Lenbok** [comment]
@kennir Can you just use `.dir-locals.el` in each project to specify a project name? e.g
```
((nil (projectile-project-name . "my-unique-project-name")))
```
(That's assuming `projectile.el`, not sure if built-in `project.el` supports variable based name overrides)

**Comment by colonelpanic8** [comment]
@Lenbok wow, quite the necropost. Last comment is from over 8 years ago! but yeah, youre suggestion should probably work.

**Comment by colonelpanic8** [comment]
Addressed on `master` in `6a95fb9`. Category mapping now follows the backend project name instead of always falling back to the project root basename, so projectile custom naming (`projectile-project-name-function` and related overrides) is respected. Closing.
